### PR TITLE
feat: Create general view interfaces

### LINKED ISSUE #307: Create general view interfaces to make view functions generic
- Create common view interfaces
- Extent View Contracts using them
- Make app methods uniform

Part of #303

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #307 

- Create generic interfaces for view
- Inherit contracts from them
- Change concrete class implementation
- Fix tests

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@mejariamol Please review

File: app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java [hunk 1]
```diff
@@ -114,25 +113,14 @@ private void listenToModelChanges() {
         attendeeListener.getNotifier()
             .filter(attendeeModelChange -> attendeeModelChange.getAction().equals(BaseModel.Action.UPDATE))
             .map(DatabaseChangeListener.ModelChange::getModel)
-            .subscribe(this::processUpdatedAttendee, Throwable::printStackTrace);
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: Seems no need to save attendees list in AttendeesPresenter now. Right?
You may remove that part too as you are on it.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: Which part?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: It is needed. To clear and add attendees

**Comment by iamareebjamal** [comment]
@arp95 @Shailesh351 @shreyasnbhat @rohanagarwal94 Please review

File: app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java (hunk 1 above)
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: >To clear and add attendees

Where?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: https://github.com/iamareebjamal/open-event-orga-app/blob/b04a696b6a81559fe14f09b691201a327eccc3c5/app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java#L96

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: I mean `attendeeList` field is of no use in the presenter now.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: I just gave you the lines where it is used

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: I am saying we are just holding attendees list in attendees presenter without any reason. Earlier it was used to update attendee after QR code scan. But now QR scan presenter is holding attendees list itself for that.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: The presenter is updating the list on receiving the changes. How is it without any reason. View should not have access to data

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: Please take a look at attendees view - presenter again. I think there is no need to save attendees list with the presenter currently.
the presenter's list is not used anywhere other than binding into the layout. which is not used for any purpose currently.
The attendees' list for the adapter is directly supplied from the `loadAttendees` result.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: I know what you are saying, but if you remove attendeeList from presenter, you'll have to add it in the view to perform these operations:

```
attendeeList.clear();
attendeeList.addAll(attendees);
```

which we don't want

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: when do we want to do these? and why?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: To update the attendee list from the repository. Haven't I already posted the code where it is used?

I think we should create a new topic or discuss on gitter. This is irrelevant to the PR

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java]: >I think we should create a new topic or discuss on gitter. This is irrelevant to the PR

yeah :smile:

**Comment by iamareebjamal** [comment]
@niranjan94 Please merge

**Pull request merged**